### ISSUE TITLE
Clarify differences between main readme and 'type files'

### ISSUE THREAD

**Issue by 0xdevalias** [ReconJSON/ReconJSON#23]
Currently it appears as though types are defined both in the main Readme, as well as in their own type file (eg `Host.md`). I feel like this leads to confusion/ambiguity, and a potential for definitions to drift between the two.

I think it would make more sense to use the main Readme to lay out 'global rules'/high level references, and leave the specific definitions to the specific 'type files'.

For example the readme would end up looking something like:

> # Structure
> A file the conforms to the ReconJSON standard...
> ...etc...
> ## Types
> The following types are defined within the ReconJSON standard:
> * `host`: This is a simple single sentence description about the `host` object (See [host](types/host.md))
> * `dns`: This is a simple single sentence description about the `dns` object (See [dns](types/dns.md))
> For each types implementation details, refer to the specific..blahblahblah

Referring to how some other specs have laid things out may help here (eg. http://openid.net/specs/openid-connect-core-1_0.html#Claims), but I think we should also consider what makes sense for a 'nice usable GitHub' layout (eg. multiple files with links), more than what the final spec layout should look like (eg. single file, multiple sections/subsections, etc)

**Comment by Rhynorater** [comment]
@0xdevalias Yeah, you're right. This is all over the place. I'll make the change.
